### ISSUE TITLE
Parrot error

### ISSUE THREAD

**Issue by Nathat23** [Nathat23/StackMob-5#132]
Errors occur when a tamed parrot is removed from your shoulder when it has no name tag.

Steps to reproduce:
1. Find a parrot
2. Tame the parrot
3. Put parrot on shoulder
4. Jump so it falls off shoulder

Code (Text):

[INDENT][20:31:11 WARN]: [StackMob] Task #8 for StackMob v5.3.3 generated an exception
java.lang.NullPointerException: null
        at uk.antiperson.stackmob.tasks.TagTask.run(TagTask.java:44) ~[?:?]
        at org.bukkit.craftbukkit.v1_16_R2.scheduler.CraftTask.run(CraftTask.java:99) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at org.bukkit.craftbukkit.v1_16_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.MinecraftServer.b(MinecraftServer.java:1354) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.DedicatedServer.b(DedicatedServer.java:376) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.MinecraftServer.a(MinecraftServer.java:1267) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:999) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_261][/INDENT]
 
Following the above error, it will also error upon death.

Code (Text):

[20:31:12 ERROR]: Could not pass event EntityDeathEvent to StackMob v5.3.3
java.lang.NullPointerException: null
        at uk.antiperson.stackmob.listeners.DeathListener.calculateDeath(DeathListener.java:68) ~[?:?]
        at uk.antiperson.stackmob.listeners.DeathListener.onStackDeath(DeathListener.java:33) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor14.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at org.bukkit.craftbukkit.v1_16_R2.event.CraftEventFactory.callEntityDeathEvent(CraftEventFactory.java:798) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.EntityLiving.d(EntityLiving.java:1490) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.EntityLiving.die(EntityLiving.java:1419) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.EntityTameableAnimal.die(SourceFile:213) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.EntityLiving.damageEntity(EntityLiving.java:1261) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.EntityParrot.damageEntity(EntityParrot.java:334) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.EntityHuman.attack(EntityHuman.java:1087) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.EntityPlayer.attack(EntityPlayer.java:1828) ~[patched_1.16.2.jar:git-Tuinity-"f663f6e"]
        at net.minecraft.server.v1_16_R2.PlayerConnection.a(PlayerConnection.java:2143) ~[patched_1.16.2.j

**Comment by Tencryn** [comment]
I compiled Master, and the null check added in `TagTask.java` did fix the first error. The second error (`Could not pass event EntityDeathEvent to StackMob`) is still present.

I was able to trigger the same two errors by creating an entity and registering it with stackmob. Note that I am not sure if this is even the right way to go about doing it, I'm just fiddling with plugins at the moment.

```java
StackEntity se = getEntityManager().registerStackedEntity((LivingEntity) entity);
se.setSize(5, true);
```
